### PR TITLE
Zer 75

### PR DESCRIPTION
**Pull request by Nazarii1017**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by andriibureviy — APPROVED** [review]

**Review by MaksymBrehin — APPROVED** [review]

File: app/controllers/admins/users_controller.rb [hunk 1]
```diff
@@ -12,10 +12,17 @@ def index
 
     def edit
       @user = User.find(params[:id])
+      render 'edit'
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:22]: remove this string, action render template with the same name automatically.  you should add `render 'template_name'` if you use another name

File: app/controllers/admins/users_controller.rb [hunk 2]
```diff
@@ -12,10 +12,17 @@ def index
 
     def edit
       @user = User.find(params[:id])
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:21]: move it to `before_action`

File: app/controllers/admins/users_controller.rb [hunk 3]
```diff
@@ -12,10 +12,17 @@ def index
...
+      render 'edit'
     end
 
     def update
       @user = User.find(params[:id])
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:26]: `before_action` and get rid of this string

File: app/controllers/admins/users_controller.rb [hunk 4]
```diff
@@ -12,10 +12,17 @@ def index
...
+      if @user.update(params[:user].permit(:first_name, :last_name, :email,
```
Comment (olegsobchuk) [inline app/controllers/admins/users_controller.rb:27]: remove `permit` to dedicated method `user_params`

File: app/models/user.rb [hunk 5]
```diff
@@ -11,10 +11,11 @@ class User < ApplicationRecord
                     length: { minimum: 6, maximum: 100 },
                     format: { with: URI::MailTo::EMAIL_REGEXP }
   validates :password, presence: true,
+                       on: :create,
```
Comment (olegsobchuk) [inline app/models/user.rb:15]: it's better to use `unless: :skip_password` and add `skip_password` `attr_reader` and use it

File: app/views/admins/users/edit.html.slim [hunk 6]
```diff
@@ -1,13 +1,68 @@
-= form_for @user do |f|
-  .field
-    = f.label :email_col
-    br
-    = f.text_field :email_col
-  .actions
-    = f.submit
+= form_for @user, url: { action: "update"} do |f|
```
Comment (olegsobchuk) [inline app/views/admins/users/edit.html.slim:1]: suppose `url: { action: "update"}` us useless here, it makes this automatically

File: app/views/admins/users/edit.html.slim [hunk 7]
```diff
@@ -1,13 +1,68 @@
...
+  - invalid = @user.errors.include?(:email)
```
Comment (olegsobchuk) [inline app/views/admins/users/edit.html.slim:2]: please consider [THIS APPROACH](https://apidock.com/rails/ActionView/Helpers/ActiveRecordHelper/error_messages_for)

File: app/views/admins/users/edit.html.slim [hunk 8]
```diff
@@ -1,13 +1,68 @@
...
+    .invalid-feedback.d-block
+      - @user.errors.full_messages_for(:country).each do |error_message|
+        = error_message
+  br
 
+  = f.submit "Update"
 
 
 
+/ = form_for(resource, as: resource_name, url: edit_admins_user_url(resource_name) do |f|
```
Comment (olegsobchuk) [inline app/views/admins/users/edit.html.slim]: remove commented code

File: config/routes.rb [hunk 9]
```diff
@@ -3,9 +3,9 @@
 Rails.application.routes.draw do
   devise_for :admins, controllers: { sessions: 'admins/sessions' }
   devise_for :users, controllers: { registrations: 'users/registrations' }
-  resources :calculators, only: %i[show edit]
+  resources :calculators, only: %i[show edit update]
```
Comment (olegsobchuk) [inline config/routes.rb:8]: `edit` and `update` should be removed from this place. User can't do changes for `calculator`

**Review by olegsobchuk — COMMENTED** [review]
Fix comments, please

**Pull request closed without merging**